### PR TITLE
 	Use a value object and a collection for log call access

### PR DESCRIPTION
**Pull request by JeroenDeDauw**
This avoids having to pass around arrays with special keys.

It also allows for adding filtering behaviour, ie

```php
$logCalls->getWithLevel( LogLevel::ERROR )->getMessages()
```

What do you think of this approach?

### CODE REVIEW

**Comment by JeroenDeDauw** [comment]
Includes https://github.com/wmde/PsrLogTestDoubles/pull/4

What do you think of this approach?

File: src/LogCalls.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+			},
+			$this->calls
+		);
+	}
+
+	/**
+	 * @return LogCall|null
+	 */
+	public function getFirstCall() {
+		return empty( $this->calls ) ? null : $this->calls[0];
```
Comment (thiemowmde) [inline src/LogCalls.php]: Is this array guaranteed to be numerically indexed, starting with 0? I assume the `LogCall... $calls` syntax does guarantee this, but I'm not familiar enough with it.

File: src/LoggerSpy.php [hunk 2]
```diff
@@ -44,17 +40,4 @@ public function assertNoLoggingCallsWhereMade() {
 		}
 	}
 
-	/**
-	 * @since 1.2
-	 * @return string[]
-	 */
-	public function getLogMessages(): array {
```
Comment (thiemowmde) [inline src/LoggerSpy.php]: I know this was never released, but this method could stay as a shortcut. Not sure if this is helpful, could also be confusing to have two ways to do the same.

**Comment by thiemowmde** [comment]
Generally looks very nice to me. :+1: But people actively working with this should decide.

File: src/LogCalls.php (hunk 1 above)
Comment (JeroenDeDauw) [inline src/LogCalls.php]: It does

**Comment by gbirke** [comment]
I like the approach, +2.

**Pull request merged**